discovery-handlers/udev/src/discovery_impl.rs

```diff
@@ -147,18 +150,13 @@ fn find_devices(
     let devices: Vec<udev::Device> = enumerator.scan_devices()?.collect();
     let final_devices = filter_by_remaining_udev_filters(devices, remaining_udev_filters);
 
-    let device_devpaths: Vec<String> = final_devices
+    let device_devpaths: Vec<DeviceProperties> = final_devices
         .into_iter()
-        .filter_map(|device| {
-            if let Some(devnode) = get_devnode(&device) {
-                Some(devnode.to_str().unwrap().to_string())
-            } else {
-                trace!(
-                    "find_devices - ignoring device with devpath {:?} due to having no devnode",
-                    get_devpath(&device)
-                );
-                None
-            }
+        .map(|device| {
+            (
+                get_devpath(&device).to_str().unwrap().to_string(),
+                get_devnode(&device).map(|devnode| devnode.to_str().unwrap().to_string()),
```

Comment:
I think something like `unwrap_or_default("")` might be good for both of these in case udev doesn't have a devpath or devnode

Reply:
nvm... looking at your example, I think rust likely does the same thing under the covers

Reply:
I agree that these are pretty, but they are transforming from `OsStr` to `String`. All devices have a devpath (`get_devpath` returns `OsStr`) while not all have `devnodes` as you pointed out (`get_devnode` returns `Option<OsStr>`)